PCF-ADR-0008-ray-cluster-authentication.md

```diff
@@ -0,0 +1,74 @@
+# Project CodeFlare - RayCluster Authentication
+|                |                                                                                  |
+| -------------- | -------------------------------------------------------------------------------- |
+| Date           | 12/19/2023                                                                       |
+| Scope          |                                                                                  |
+| Status         | provisional                                                                      |
+| Authors        | [Kevin Postlethwait](@KPostOffice)                                               |
+| Supersedes     | N/A                                                                              |
+| Superseded by: | N/A                                                                              |
+| Issues         |                                                                                  |
+| Other docs:    | none                                                                             |
+
+## What
+
+Enabling OpenShift OAuth as an identity provider and authorization mechanism for KubeRay RayClusters.
+
+## Why
+
+Currently there is no authentication by default enabled in front of the Ray Dashboard and we would like to offer a
+unified login experience for the whole of ODH by way of OpenShift OAuth
+
+## Goals
+
+* Protect the dashboard from unauthenticated access by way of an OAuth flow
+* Protect the API endpoint from unauthenticated access by way of bearer tokens
+* Enable HTTPS by default for accessing the RayCluster API and Dashboard
+
+## Non-Goals
+
+* Creating a fully customizable auth(z) solution
+
+## How
+
+In order to solve this we will add a RayCluster reconciler which creates the necessary resources for enabling
+enabling authentication and authorization in front of the API and dashboard.
+
+![Auth Diagram](images/PCF-ADR-008-ray-cluster-auth.jpg)
+
+This will change in Q1 or early Q2 of 2024 with the introduction of Authorino. The overall design stays the same, but
+the resources created by the CodeFlare Operator's RayCluster reconciler will change and there will likely no longer be
+a need for adding the OAuth Proxy sidecar to the head node definition.
+
+## Blockers
+
+* Currently KubeRay is using `update` rather than `patch` to update the CRs status. This may cause the two reconcilers
```

Comment:
Is the CFO controller really going to update the RayCluster status sub-resource? Or is it only updating the spec?

Reply:
My only thought here was that it would be nice to report the status of the resources the the CFO creates in the RayCluster status field. Do you have another suggestion for how to make the status discoverable?

Reply:
Right, that would be ideal to report useful information in a status condition. That'll require to update the RayCluster API though. Another, less elegant approach would be to use annotations.